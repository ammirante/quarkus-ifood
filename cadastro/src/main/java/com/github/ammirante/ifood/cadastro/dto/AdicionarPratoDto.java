package com.github.ammirante.ifood.cadastro.dto;

import java.math.BigDecimal;

/**
 * AdicionarPratoDto
 *
 */
public class AdicionarPratoDto extends CadastroDTO {

	/**
	 * 
	 */
	private static final long serialVersionUID = -3086611945955899423L;
	
	public String nome;
	public String descricao;
	public BigDecimal preco;
	
}
