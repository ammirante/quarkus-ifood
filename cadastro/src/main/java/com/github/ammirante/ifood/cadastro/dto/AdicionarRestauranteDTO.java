package com.github.ammirante.ifood.cadastro.dto;

/**
 * AdicionarRestauranteDTO
 *
 */
public class AdicionarRestauranteDTO extends CadastroDTO {

	/**
	 * 
	 */
	private static final long serialVersionUID = 2689633935434853053L;
	
	public String proprietario;
	public String cnpj;
	public String nomeFantasia;
	public LocalizacaoDTO localizacaoDTO;
	
}
