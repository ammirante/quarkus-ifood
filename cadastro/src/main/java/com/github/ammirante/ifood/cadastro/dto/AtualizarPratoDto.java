package com.github.ammirante.ifood.cadastro.dto;

import java.io.Serializable;

/**
 * AtualizarPratoDto
 *
 */
public class AtualizarPratoDto implements Serializable {
	
	/**
	 * 
	 */
	private static final long serialVersionUID = -7860001516826682877L;
	
	private String nome;
	
	/**
	 * @return the nome
	 */
	public String getNome() {
		return nome;
	}

	/**
	 * @param nome the nome to set
	 */
	public void setNome(String nome) {
		this.nome = nome;
	}
}
