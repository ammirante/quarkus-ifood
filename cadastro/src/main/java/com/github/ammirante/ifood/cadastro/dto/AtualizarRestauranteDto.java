package com.github.ammirante.ifood.cadastro.dto;

import java.io.Serializable;

/**
 * AtualizarRestauranteDto
 *
 */
public class AtualizarRestauranteDto implements Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = -1951876794590550707L;
	
	private String nome;
	
	/**
	 * @return the nome
	 */
	public String getNome() {
		return nome;
	}

	/**
	 * @param nome the nome to set
	 */
	public void setNome(String nome) {
		this.nome = nome;
	}
}
