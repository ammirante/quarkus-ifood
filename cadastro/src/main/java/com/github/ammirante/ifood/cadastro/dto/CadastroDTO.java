package com.github.ammirante.ifood.cadastro.dto;

import java.io.Serializable;

import com.github.ammirante.ifood.cadastro.util.ToString;

/**
 * Classe padrão para utilização de DTO.
 *
 */
public class CadastroDTO implements Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = 2055796904694797729L;

	/**
	 *
	 */
	public String toString() {
		return ToString.build(this);
	}
}
