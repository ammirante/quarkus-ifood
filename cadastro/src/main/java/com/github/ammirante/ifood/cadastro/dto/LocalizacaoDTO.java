package com.github.ammirante.ifood.cadastro.dto;

/**
 * LocalizacaoDTO
 *
 */
public class LocalizacaoDTO extends CadastroDTO {
	
	/**
	 * 
	 */
	private static final long serialVersionUID = 4310774226303642715L;
	
	public Double latitute;
	public Double longitude;
	
}
