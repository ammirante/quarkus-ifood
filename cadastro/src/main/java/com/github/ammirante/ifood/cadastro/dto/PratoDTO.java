package com.github.ammirante.ifood.cadastro.dto;

/**
 * PratoDTO
 *
 */
public class PratoDTO extends CadastroDTO {
	
	/**
	 * 
	 */
	private static final long serialVersionUID = -608004667868887716L;
	
	public Long id;
	public String nome;
	public String descricao;
	public String preco;
	public RestauranteDTO restaurante;
	
}
