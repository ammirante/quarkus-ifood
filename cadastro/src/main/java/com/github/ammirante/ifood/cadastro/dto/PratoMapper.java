package com.github.ammirante.ifood.cadastro.dto;

import org.mapstruct.Mapper;
import org.mapstruct.MappingTarget;

import com.github.ammirante.ifood.cadastro.entidade.Prato;

/**
 * PratoMapper
 *
 */
@Mapper(componentModel = "cdi")
public interface PratoMapper {

	/**
	 * @param dto
	 * @return
	 */
	public Prato toPrato(AdicionarPratoDto dto);
	
	/**
	 * @param dto
	 * @param prato
	 */
	public void toPrato(AtualizarPratoDto dto, @MappingTarget Prato prato);
	
	/**
	 * @param prato
	 * @return
	 */
	public PratoDTO toPratoDTO(Prato prato);
	
}
