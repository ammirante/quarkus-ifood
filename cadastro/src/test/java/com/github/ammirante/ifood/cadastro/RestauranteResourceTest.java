package com.github.ammirante.ifood.cadastro;

import static io.restassured.RestAssured.given;

import java.util.Optional;

import javax.ws.rs.core.Response.Status;

import org.approvaltests.Approvals;
import org.junit.Assert;
import org.junit.jupiter.api.Test;

import com.github.ammirante.ifood.cadastro.dto.AtualizarPratoDto;
import com.github.ammirante.ifood.cadastro.dto.AtualizarRestauranteDto;
import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;

import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.http.ContentType;

/**
 * RestauranteResourceTest
 *
 */
@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CadastroTestLifecycleManager.class)
public class RestauranteResourceTest {

	private static final Long ID_RESTAURANTE = 123L;
	private static final Long ID_PRATO = 123L;
	
    /**
     * 
     */
    @Test
    @DataSet(value = "restaurantes-cenario-1.yml")
    public void testBuscarRestaurantes() {
    	String resultado = given()
    			.when()
    			.get("/restaurantes")
    			.then()
    			.statusCode(Status.OK.getStatusCode())
    			.extract()
    			.asString()
    			;
    	Approvals.verifyJson(resultado);
    }
    
    /**
     * 
     */
    @Test
    @DataSet(value = "restaurantes-cenario-1.yml")
    public void testAtualizarRestaurante() {
    	AtualizarRestauranteDto atualizarRestauranteDto = new AtualizarRestauranteDto();
    	atualizarRestauranteDto.setNome("Novo nome");
    	given()
    	.with()
    	.contentType(ContentType.JSON)
    	.pathParam("id", ID_RESTAURANTE)
    	.body(atualizarRestauranteDto)
    	.when()
    	.put("/restaurantes/{id}")
    	.then()
    	.statusCode(Status.NO_CONTENT.getStatusCode())
    	;
    	
    	Restaurante restaurante = Restaurante.findById(ID_RESTAURANTE);
    	Assert.assertEquals(atualizarRestauranteDto.getNome(), restaurante.nome);
    }
    
    @Test
    @DataSet(value = "restaurantes-cenario-1.yml", cleanAfter = true)    
    public void testDeletarRestaurante() {
    	given()
    	.with()
    	.pathParam("id", ID_RESTAURANTE)
    	.when()
    	.delete("/restaurantes/{id}")
    	.then()
    	.statusCode(Status.NO_CONTENT.getStatusCode())
    	;
    	
    	Optional<Restaurante> restaurante = Restaurante.findByIdOptional(ID_RESTAURANTE);
    	Assert.assertTrue(restaurante.isEmpty());
    }
    
    /**
     * 
     */
    @Test
    public void testAdicionarRestaurante() {
    	Restaurante restauranteInsercao = new Restaurante();
    	restauranteInsercao.nome = "Novo restaurante";
    	given()
    	.contentType(ContentType.JSON)
    	.body(restauranteInsercao)
    	.when()
    	.post("/restaurantes")
    	.then()
    	.statusCode(Status.CREATED.getStatusCode())
    	;
    }
    
    /**
     * 
     */
    @Test
    @DataSet(value = "pratos-cenario-1.yml", cleanAfter = true)
    public void testBuscarPratos() {
    	String resultado = given()
    			.when()
    			.with()
    			.pathParam("idRestaurante", ID_RESTAURANTE)
    			.get("/restaurantes/{idRestaurante}/pratos")    			
    			.then()
    			.statusCode(Status.OK.getStatusCode())
    			.extract()
    			.asString()
    			;
    	Approvals.verifyJson(resultado);
    }
    
    /**
     * 
     */
    @Test
    @DataSet(value = "pratos-cenario-1.yml")
    public void testAtualizarPrato() {
    	AtualizarPratoDto atualizarPratoDto = new AtualizarPratoDto();
    	atualizarPratoDto.setNome("Novo nome");
    	given()
    	.with()
    	.contentType(ContentType.JSON)
    	.pathParam("idRestaurante", ID_RESTAURANTE)
    	.pathParam("id", ID_PRATO)
    	.body(atualizarPratoDto)
    	.when()
    	.put("/restaurantes/{idRestaurante}/pratos/{id}")
    	.then()
    	.statusCode(Status.NO_CONTENT.getStatusCode())
    	;
    	
    	Prato prato = Prato.findById(ID_PRATO);
    	Assert.assertEquals(atualizarPratoDto.getNome(), prato.nome);
    }
    
    /**
     * 
     */
    @Test
    @DataSet(value = "pratos-cenario-1.yml", cleanAfter = true)    
    public void testDeletarPrato() {
    	given()
    	.with()
    	.pathParam("idRestaurante", ID_RESTAURANTE)
    	.pathParam("id", ID_PRATO)
    	.when()
    	.delete("/restaurantes/{idRestaurante}/pratos/{id}")
    	.then()
    	.statusCode(Status.NO_CONTENT.getStatusCode())
    	;
    	
    	Optional<Prato> prato = Prato.findByIdOptional(ID_PRATO);
    	Assert.assertTrue(prato.isEmpty());
    }
    
    /**
     * 
     */
    @Test
    public void testAdicionarPrato() {
    	Prato pratoInsercao = new Prato();
    	pratoInsercao.nome = "Novo prato";
    	given()
    	.contentType(ContentType.JSON)
    	.pathParam("idRestaurante", ID_RESTAURANTE)
    	.body(pratoInsercao)
    	.when()
    	.post("/restaurantes/{idRestaurante}/pratos")
    	.then()
    	.statusCode(Status.CREATED.getStatusCode())
    	;
    }

}