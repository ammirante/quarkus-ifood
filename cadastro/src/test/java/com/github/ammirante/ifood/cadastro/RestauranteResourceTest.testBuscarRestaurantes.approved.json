[
  {
    "id": 123,
    "nome": "Novo Restaurante",
    "proprietario": "ID do keyclock"
  }
]